From e757b2191494c8acad09a5dda001e649cdc62d0f Mon Sep 17 00:00:00 2001
From: "tao.dong" <tao.dong@amlogic.com>
Date: Tue, 17 Jun 2014 10:36:06 +0800
Subject: [PATCH 4354/5965] Revert this change by zhanghui "pd# 87592: skip
 first frame if it was non-I frame  because of it might be blocky in vmpeg12"

This reverts commit 577c4211c76df1d6aaf35ac3ea6f13f0fd3ea2e0.
---
 drivers/amlogic/amports/vmpeg12.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/amlogic/amports/vmpeg12.c b/drivers/amlogic/amports/vmpeg12.c
index 2e3689b80bf0..69dfacc7da7b 100755
--- a/drivers/amlogic/amports/vmpeg12.c
+++ b/drivers/amlogic/amports/vmpeg12.c
@@ -120,7 +120,7 @@ static int vmpeg_event_cb(int type, void *data, void *private_data);
 
 static void vmpeg12_prot_init(void);
 static void vmpeg12_local_init(void);
-static u32 frame_counter = 0;
+
 static const char vmpeg12_dec_id[] = "vmpeg12-dev";
 #define PROVIDER_NAME   "decoder.mpeg12"
 static const struct vframe_operations_s vmpeg_vf_provider =
@@ -371,6 +371,7 @@ static irqreturn_t vmpeg12_isr(int irq, void *dev_id)
             vf = vfq_pop(&newframe_q);
 
             vfbuf_use[index] = 2;
+
             set_frame_info(vf);
 
             vf->index = index;
@@ -387,11 +388,9 @@ static irqreturn_t vmpeg12_isr(int irq, void *dev_id)
             vf->canvas0Addr = vf->canvas1Addr = index2canvas(index);
             vf->pts = (pts_valid) ? pts : 0;
 
-            if (error_skip(info, vf)|| !frame_counter && !((info & PICINFO_TYPE_MASK) == PICINFO_TYPE_I)) {
+            if (error_skip(info, vf)) {
                 vfq_push(&recycle_q, vf);
-
             } else {
-              //  frame_counter = 1;
                 vfq_push(&display_q, vf);
                 vf_notify_receiver(PROVIDER_NAME,VFRAME_EVENT_PROVIDER_VFRAME_READY,NULL);
             }
@@ -414,8 +413,8 @@ static irqreturn_t vmpeg12_isr(int irq, void *dev_id)
             vf->orientation = 0 ;
             vf->canvas0Addr = vf->canvas1Addr = index2canvas(index);
             vf->pts = 0;
-            if (error_skip(info, vf)|| !frame_counter&& !((info & PICINFO_TYPE_MASK) == PICINFO_TYPE_I)) {
-                frame_counter = 1;
+
+            if (error_skip(info, vf)) {
                 vfq_push(&recycle_q, vf);
             } else {
                 vfq_push(&display_q, vf);
@@ -739,7 +738,7 @@ static void vmpeg12_local_init(void)
 
     frame_force_skip_flag = 0;
     wait_buffer_counter = 0;
-    frame_counter = 0;
+
     dec_control &= DEC_CONTROL_INTERNAL_MASK;
 }
 
-- 
2.19.0

